cabal-install/Distribution/Client/ProjectConfig/Legacy.hs

```diff
@@ -841,6 +843,7 @@ legacySharedConfigFieldDescrs =
       , "one-shot", "jobs", "keep-going", "offline"
         -- solver flags:
       , "max-backjumps", "reorder-goals", "count-conflicts", "strong-flags"
+      , "index-snapshot"
```

Comment:
Bike-shedding opportunity: what should the user facing name for this feature be? I'm not sure "index" helps. In the config files we call them "repository" so perhaps "repository-snapshot" is good. It'd also be nice to allow proper human readable timestamps in the config files, e.g. in RFC format rather than unix time number.